tests/tests_app/core/test_lightning_app.py

```diff
@@ -884,3 +887,56 @@ def test_slow_flow():
     )
 
     MultiProcessRuntime(app1).dispatch()
+
+
+class SizeWork(LightningWork):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.counter = 0
+
+    def run(self, signal: int):
+        self.counter += 1
+
+
+class SizeFlow(LightningFlow):
+    def __init__(self):
+        super().__init__()
+        self.work0 = SizeWork(parallel=True, cache_calls=True)
+        self._state_sizes = {}
+
+    def run(self):
+        for idx in range(self.work0.counter + 2):
+            self.work0.run(idx)
+
+        self._state_sizes[self.work0.counter] = asizeof.asizeof(self.state)
+
+        if self.work0.counter >= 20:
+            self._exit()
+
+
+def test_state_size_is_bounded():
```

Comment:
Not testing it's bounded, rather that it grows slowly (also see final review comment for proposal on how to make it effectively bounded or close to it).

Reply:
Maybe a good test would also be looking at the size of the circulating state, to test that what we communicate is bounded.

Reply:
Done !